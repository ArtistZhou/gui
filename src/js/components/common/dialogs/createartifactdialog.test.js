import React from 'react';
import renderer from 'react-test-renderer';
import CreateArtifactDialog from './createartifactdialog';

it('renders correctly', () => {
  const tree = renderer.create(<CreateArtifactDialog />).toJSON();
  expect(tree).toMatchSnapshot();
});
