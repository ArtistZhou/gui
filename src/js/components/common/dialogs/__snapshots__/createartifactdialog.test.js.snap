// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateArtifactDialog Component renders correctly 1`] = `
"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthMd MuiDialog-paperFullWidth MuiPaper-elevation24 MuiPaper-rounded\\" role=\\"dialog\\"><div class=\\"MuiDialogTitle-root\\"><h2 class=\\"MuiTypography-root MuiTypography-h6\\">Creating a new Artifact</h2></div><div class=\\"MuiDialogContent-root onboard-dialog dialog-content\\"><div><div class=\\"muted\\">Follow these steps on your workstation. Estimated time 5 minutes.</div><ol class=\\"spaced-list\\"><li>Download both mender-artifact and single-file-artifact-gen and make them executable by running:<div class=\\"code\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\" style=\\"float: right; margin: -10px 0px 0px 10px;\\"><span class=\\"MuiButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z\\"></path></svg>Copy to clipboard</span><span class=\\"MuiTouchRipple-root\\"></span></button><span style=\\"word-break: break-word;\\">
wget https://d1b0l86ne08fsf.cloudfront.net/mender-artifact/master/linux/mender-artifact &amp;&amp; \\\\
chmod +x mender-artifact &amp;&amp; \\\\
wget https://raw.githubusercontent.com/mendersoftware/mender/master/support/modules-artifact-gen/single-file-artifact-gen &amp;&amp; \\\\
chmod +x single-file-artifact-gen &amp;&amp; \\\\
sudo cp mender-artifact single-file-artifact-gen /usr/local/bin/
</span></div><p></p></li><li>Next, create a new <i>index.html</i> file with the simple contents 'Hello world'. This will be the web page of your updated application, so you'll be able to easily see when your device has received the update. Copy and run the command:<div class=\\"code\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\" style=\\"float: right; margin: -20px 0px 0px 10px;\\"><span class=\\"MuiButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z\\"></path></svg>Copy to clipboard</span><span class=\\"MuiTouchRipple-root\\"></span></button><span style=\\"word-break: break-word;\\">cat &gt;index.html &lt;&lt;EOF
Hello World!
EOF
</span></div><p></p></li><li>Now you can create a new version of the demo webserver application with this <i>index.html</i> file. Generate a new Artifact by copying &amp; running:<div class=\\"code\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\" style=\\"float: right; margin: -10px 0px 0px 10px;\\"><span class=\\"MuiButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z\\"></path></svg>Copy to clipboard</span><span class=\\"MuiTouchRipple-root\\"></span></button><span style=\\"word-break: break-word;\\">
ARTIFACT_NAME=\\"demo-webserver-updated\\" &amp;&amp; \\\\
DEVICE_TYPE=\\"qemux86-64\\" &amp;&amp; \\\\
OUTPUT_PATH=\\"demo-webserver-updated.mender\\" &amp;&amp; \\\\
DEST_DIR=\\"/var/www/localhost/htdocs/\\" &amp;&amp; \\\\
FILE_NAME=\\"index.html\\" &amp;&amp; \\\\
single-file-artifact-gen -n \${ARTIFACT_NAME} -t \${DEVICE_TYPE} -d \${DEST_DIR} -o \${OUTPUT_PATH} \${FILE_NAME}
</span></div><p></p></li></ol></div></div><div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Cancel</span><span class=\\"MuiTouchRipple-root\\"></span></button><div style=\\"flex-grow: 1;\\"></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Next</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"
`;
